### PR TITLE
`azurerm_backup_protected_vm` - split create and update function to fix lifecycle - ignore

### PR DESCRIPTION
**Pull request by mbfrahry**

### CODE REVIEW

File: internal/services/recoveryservices/backup_protected_vm_resource.go [hunk 1]
```diff
@@ -59,22 +59,20 @@ func resourceRecoveryServicesBackupProtectedVM() *pluginsdk.Resource {
...
 
 	vaultName := d.Get("recovery_vault_name").(string)
 
 	// source_vm_id must be specified at creation time but can be removed during update
-	if d.IsNewResource() {
-		if _, ok := d.GetOk("source_vm_id"); !ok {
-			return fmt.Errorf("`source_vm_id` must be specified when creating")
-		}
+	if _, ok := d.GetOk("source_vm_id"); !ok {
+		return fmt.Errorf("`source_vm_id` must be specified when creating")
```
Comment (stephybun) [inline internal/services/recoveryservices/backup_protected_vm_resource.go:75]: I can't comment on the line, but there's a `d.IsNewResource()` check on line 81 which can be removed as well.

File: internal/services/recoveryservices/backup_protected_vm_resource.go [hunk 2]
```diff
@@ -145,27 +140,23 @@ func resourceRecoveryServicesBackupProtectedVMCreateUpdate(d *pluginsdk.Resource
...
-			return fmt.Errorf("creating/updating Azure Backup Protected VM %q (Resource Group %q): %+v", protectedItemName, resourceGroup, err)
-		}
 
-		operationId, err := parseBackupOperationId(resp.HttpResponse)
-		if err != nil {
-			return fmt.Errorf("issuing creating/updating request for %s: %+v", id, err)
-		}
+	resp, err := client.CreateOrUpdate(ctx, id, item)
+	if err != nil {
+		return fmt.Errorf("creating Azure Backup Protected VM %q (Resource Group %q): %+v", protectedItemName, resourceGroup, err)
```
Comment (stephybun) [inline internal/services/recoveryservices/backup_protected_vm_resource.go:146]: Can we not use the id for the error message here?
```suggestion
		return fmt.Errorf("creating %s: %+v", id, err)
```

File: internal/services/recoveryservices/backup_protected_vm_resource.go [hunk 3]
```diff
@@ -145,27 +140,23 @@ func resourceRecoveryServicesBackupProtectedVMCreateUpdate(d *pluginsdk.Resource
...
+	}
 
-		if err = resourceRecoveryServicesBackupProtectedVMWaitForStateCreateUpdate(ctx, opClient, id, operationId); err != nil {
-			return err
-		}
+	operationId, err := parseBackupOperationId(resp.HttpResponse)
+	if err != nil {
+		return fmt.Errorf("issuing creating request for %s: %+v", id, err)
```
Comment (stephybun) [inline internal/services/recoveryservices/backup_protected_vm_resource.go:151]:
```suggestion
		return fmt.Errorf("issuing create request for %s: %+v", id, err)
```

File: internal/services/recoveryservices/backup_protected_vm_resource.go [hunk 4]
```diff
@@ -176,14 +167,14 @@ func resourceRecoveryServicesBackupProtectedVMCreateUpdate(d *pluginsdk.Resource
...
 		if err != nil {
-			return fmt.Errorf("creating/updating %s: %+v", id, err)
+			return fmt.Errorf("creating %s: %+v", id, err)
 		}
 
-		operationId, err := parseBackupOperationId(resp.HttpResponse)
+		operationId, err = parseBackupOperationId(resp.HttpResponse)
 		if err != nil {
-			return fmt.Errorf("issuing creating/updating request for %s: %+v", id, err)
+			return fmt.Errorf("issuing creating request for %s: %+v", id, err)
```
Comment (stephybun) [inline internal/services/recoveryservices/backup_protected_vm_resource.go:177]:
```suggestion
			return fmt.Errorf("issuing create request for %s: %+v", id, err)
```

**Review by stephybun — APPROVED** [review]
Thanks @mbfrahry, a few minor comments otherwise LGTM provided the tests are passing 👍

**Comment by mbfrahry** [comment]
Thanks @stephybun! Took care of the comments and tests are passing!

**Pull request merged**